examples/mnist_export_v2/export_test.py

```diff
@@ -0,0 +1,57 @@
+import tensorflow as tf
+import tensorflow_hub as hub
+import unittest
+import tempfile
+import os
+from absl.testing import absltest
+import export
+
+model = None
+TEST_DATA_DIR = "examples/mnist_export_v2/test_data"
+
+def load_model(path):
```

Comment:
I understand the speed reduction, but loading should be extremely fast in this and doesn't justify adding a confusing global variable.

Also, what if we want to export and load a different model every time?

Reply:
For this test we are looking for one trained model. The idea is, it will train the model in `test_model_exporting`. And this trained model will be loaded for checking if this can actually be used for inferencing. If I remove this, it will train the model on the same dataset, thrice for three different steps, just to obtain  th trained graph. Once for exporting, once for testing on empty_input, and finally once for inferencing.

Reply:
I'm not really sure, if removing this would help.

Reply:
Yes, I see. It looks like a not very good design to do the following things:
1. call one test from another,
2. make one test call produce artifacts used by a different test.

It's fine to create it multiple times. You can create it the smallest possible and train for 1 step.

Reply:
Do you think training for one step would produce enough updates to run the inference test?

Reply:
Hey @vbardiovskyg , I just found out, since I've been using `tmpfile.mkdtemp()`, there was this unavoidable need to use `hub.load()`. since after you asked me to use `self.get_temp_dir()`, this problem of file getting deleted after each step, is solved! So, removing that `load_model()` function would work just fine! Thanks :)

Reply:
Honestly, I think the most reasonable way is to not test the trained model for whether it can predict an image with reasonable accuracy.

Instead I would just test for formal correctness, meaning that the code is syntactically right, etc. This way you don't need to include any test data / images and we will save a lot of resources. At the risk of messing up this script in the future without noticing, but that risk is extremely low.

Reply:
So, what do you suggest? should I remove that? I noticed something similar in text_embeddings
https://github.com/tensorflow/hub/blob/caa9e03f3b483fa611efd49a81b5c512ef62a6dc/examples/text_embeddings/export_test.py#L66-L84
So, I thought, why not test if the model is even producing something which actually makes sense. If you suggest, I'll remove this test :), but I think, being a small enough model, and having a test data of just a few kilobytes, it shouldn't consume too much of resource. and also bazel test currently takes 57 secs to run.

Reply:
The difference is that exporter in text_embeddings is not training anything, just building a simple graph.

I think it is totally fine to test training on a couple of images and then run inference. I would even recommend that. But I don't think it is fine to train a model fully to achieve a predictable performance.

Reply:
I'm actually training for one epoch, and with big enough `batch_size` to speed up the process. I thnk for the code, that's the least amount of training I can perform. I need to dig deep to find a better way, if possible.